v0/destinations/hs/util.js

```diff
@@ -532,23 +464,91 @@ const getExistingData = async (inputs, destination) => {
       );
     }
 
-    const after = searchResponse.response?.paging?.next?.after | 0;
+    const after = searchResponse.response?.paging?.next?.after || 0;
 
     requestData.after = after; // assigning to the new value of after
     checkAfter = after; // assigning to the new value if no after we assign it to 0 and no more calls will take place
 
     const results = searchResponse.response?.results;
     if (results) {
-      results.map(result => {
+      updateHubspotIds = results.map(result => {
         const propertyValue = result.properties[identifierType];
-        updateHubspotIds.push({ id: result.id, property: propertyValue });
+        return { id: result.id, property: propertyValue };
       });
     }
   }
-
   return updateHubspotIds;
 };
 
+const setHsSearchId = (input, id) => {
+  const { message } = input;
+  const resultExternalId = [];
+  const externalIdArray = message.context?.externalId;
+  if (externalIdArray) {
+    externalIdArray.forEach(extIdObj => {
+      const { type } = extIdObj;
+      if (type.includes("HS")) {
+        extIdObj.hsSearchId = id;
+      }
+      resultExternalId.push(extIdObj);
+    });
+  }
+  return resultExternalId;
+};
+
+/**
+ *
+ * To reduce the number of calls for searching of already existing objects
+ * We do search for all the objects before router transform and assign the type (create/update)
+ * accordingly to context.hubspotOperation
+ *
+ * */
+
+const splitEventsForCreateUpdate = async (inputs, destination) => {
+  // get all the id and properties of already existing objects needed for update.
+  const updateHubspotIds = await getExistingData(inputs, destination);
+
+  const resultInput = inputs.map(input => {
+    const { message } = input;
+    const { destinationExternalId } = getDestinationExternalIDInfoForRetl(
+      message,
+      "HS"
+    );
+
+    const filteredInfo = updateHubspotIds.filter(
+      update => update.property.toString() === destinationExternalId.toString()
```

Comment:
so `===` is causing the trouble  as the hubspot converts everything to string and rudder-sources we retain the datatype? right?

Reply:
yes